app/controllers/admin/admin_users_controller.rb

```diff
@@ -247,6 +247,18 @@ def send_notification
     redirect_to :action => :notify
   end
 
+  def check
+    @user = User.find_by_login(params[:id])
+    @user.fix_user_subscriptions
```

Comment:
I think shalott had the right idea about renaming this action "troubleshoot" rather than "check" -- we're fixing rather than merely observing what's wrong (checking). The success message could be something like "User account troubleshooting complete." 

Also, there's a discrepancy in how the methods are named -- some use `users` and others `user`. I'd suggest going with `user` on all of them to avoid any confusion where `users` is assumed to be a plural rather than a possessive. 

For `check_users_work_dates`, I'd again suggest a verb other than `check` -- maybe `set_user_work_dates`, since we're setting work dates where it's currently nil.